pkg/utils/test_util.go

```diff
@@ -177,3 +185,64 @@ func (m TestMapper) RESTMapping(gk schema.GroupKind, _ ...string) (*meta.RESTMap
 	}
 	return nil, errors.New("test error: mapping does not exist")
 }
+
+// CreateMockAttributeBasedVMSizeRecommenderServer creates a mock HTTP server for testing AttributeBasedVMSizeRecommenderClient.
```

Comment:
it should be under test/utils folder. This file should be deprecated and we cannot mix the test code and prod code.

https://github.com/Azure/fleet/blob/main/pkg/clients/azure/compute/vmsizerecommenderclient_test.go#L208 Can you also refactor this code too?